crates/core/tedge_mapper/src/aws/converter.rs

```diff
@@ -44,13 +49,40 @@ impl Converter for AwsConverter {
     }
 
     async fn try_convert(&mut self, input: &Message) -> Result<Vec<Message>, Self::Error> {
-        let () = self.size_threshold.validate(input)?;
+        self.size_threshold.validate(input)?;
         let default_timestamp = self.add_timestamp.then(|| self.clock.now());
-        let mut serializer = ThinEdgeJsonSerializer::new_with_timestamp(default_timestamp);
-        let () = tedge_api::parser::parse_str(input.payload_str()?, &mut serializer)?;
 
-        let payload = serializer.into_string()?;
-        Ok(vec![(Message::new(&self.mapper_config.out_topic, payload))])
+        // serialize with ThinEdgeJson for measurements, for alarms and events just add the timestamp
+        let payload = if input.topic.name.starts_with("tedge/measurements") {
+            let mut serializer = ThinEdgeJsonSerializer::new_with_timestamp(default_timestamp);
+            tedge_api::parser::parse_str(input.payload_str()?, &mut serializer)?;
+
+            serializer.into_string()?
+        } else {
+            let mut payload_json: Map<String, Value> = serde_json::from_slice(&input.payload)?;
+
+            if let Some(timestamp) = default_timestamp {
+                let timestamp = timestamp
+                    .format(&time::format_description::well_known::Rfc3339)?
+                    .as_str()
+                    .into();
+                payload_json.entry("time").or_insert(timestamp);
+            }
+
+            serde_json::to_string(&payload_json)?
+        };
+
+        // topic guaranteed to be tedge/measurements/..., tedge/alarms/..., or tedge/events/...
```

Comment:
Unfortunately there is no such guarantees. Worse some messages sent on some topics under `tedge/#` are not meant for the cloud but for the child devices. See for instance [`tedge/health-check`](https://thin-edge.github.io/thin-edge.io/html/howto-guides/021_enable_tedge_watchdog_using_systemd.html?highlight=tedge%2F#step-2-start-the-tedge-watchdog-service) or [`tedge/configuration_change`](https://thin-edge.github.io/thin-edge.io/html/references/c8y-configuration-management.html?highlight=tedge%2Fconfiguration#notifications).

* @Bravo555 => You need to add an explicit filter to uniquely match measurements, alarms and events.
* @reubenmiller => We also need to address this kind of issues while designing the new scheme for MQTT topics.

Reply:
Only `tedge/measurements/`, `tedge/alarms/` and `tedge/events/` are parts of the input topic filter though, so I thought that only these topics will be subscribed by the mapper.

Reply:
Ah, okay, nevermind, I see that more subscriptions get added later by the calling code. For the time being I'll add proper `if` checks to this function.

Reply:
@didier-wenzek added the checks